pkg/operator/reports.go

```diff
@@ -734,6 +757,58 @@ func (op *Reporting) runReport(logger log.FieldLogger, report *metering.Report)
 	return nil
 }
 
+func isReportNotUsedAsInput(report *metering.Report, op *Reporting) bool {
+	// Consider Reports referencing this report in the namespace of this report
+	reports, err := op.reportLister.Reports(report.Namespace).List(labels.Everything())
+	if err != nil {
+		op.logger.Errorf("unable to determine list of reports that might be input for report: %s", report.Name)
+		return false
+	}
+	for _, report := range reports {
+		// If Report has no spec.inputs it can't be depending on this report.
+		if report.Spec.Inputs == nil {
+			continue
+		}
+
+		if op.doReportsDependOnThisReport(report) {
+			op.logger.Infof("reports exist that use report %s as input, will not delete though retention period has expired", report.Name)
+			return false
+		}
```

Comment:
Can we make this clear in the description of this function that we exit early if we detect that the Report we're currently processing has a dependency on another Report, we stop processing further and exit early? I missed that the first time around and assumed we would just then process the ReportQueries next, even if we detected a report dependency in the previous block.

Reply:
adding function comment